### PR TITLE
Tooltip Component for SaaSquatch Portal

### PR DESCRIPTION
**Pull request by kutaycinar**
## Description of the change

image

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

 - Jira issue number: (PUT IT HERE)
 - Process.st launch checklist: (PUT IT HERE)

### Development

- [ ] [Prettier](https://www.npmjs.com/package/prettier) was run
- [ ] The behaviour changes in the pull request are covered by specs
- [ ] All tests related to the changed code pass in development

### Paperwork

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has a Jira number
- [ ] This pull request has a Process.st launch checklist
- [ ] "Ready for review" label attached to the PR and reviewers pinged on Slack

### Code review 

- [ ] Changes have been reviewed by at least one other engineer
- [ ] Security impacts of this change have been considered

### CODE REVIEW

File: packages/visual-dev/src/components/stories/Tooltip.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+export const Right = () => (
+	<StoryFrame>
+		<Tooltip
+			direction="right"
+			text="A tooltip that explains something in detail"
+		>
+			<button> Hover </button>
+		</Tooltip>
+	</StoryFrame>
+)
```
Comment (noahwc) [inline packages/visual-dev/src/components/stories/Tooltip.stories.tsx:61]: There's a couple states I'd like to see added as stories:
- Really long text (like multiple sentences). In the designs it looks like there's a max width for tooltips before text starts wrapping
- No `direction` value is passed and the component falls back to default
- An empty string is passed as text

**Review by noahwc — CHANGES_REQUESTED** [review]

**Review by Locrian24 — CHANGES_REQUESTED** [review]
Component as a whole is great, just need to split the styling into seperate styled components instead of using classes and then re-using `TooltipDiv` (anti-pattern)

File: packages/visual-dev/src/components/Tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	direction = "top",
+	children
+}) => {
+	let delay: ReturnType<typeof setTimeout>;
+	const [active, setActive] = useState(false);
+
+	const showTooltip = () => {
+		delay = setTimeout(() => {
+			setActive(true);
+		}, 250);
```
Comment (Locrian24) [inline packages/visual-dev/src/components/Tooltip/Tooltip.tsx:32]: Having a delay on the tooltip without any animation feels weird when hovering. Generally, tooltips show up pretty much instantly once the user has hovered. Please lower the delay to something less noticeable or remove delay completely.

**Review by Locrian24 — CHANGES_REQUESTED** [review]
Delay on the tooltip is a bit too long, causing it to feel weird since it shows up so late. Please lower or remove

File: packages/visual-dev/src/components/Tooltip/Tooltip.tsx (hunk 2 above)
Comment (kutaycinar) [reply packages/visual-dev/src/components/Tooltip/Tooltip.tsx:32]: I experimented with using no delay initially but found it to feel unnatural. Tooltips typically do have a bit of delay before showing on screen. You can see this when hovering over chrome tabs, photoshop tools, or even the windows start menu.

_Tooltips ... are other examples of common elements with hover interactions that can prove annoying to users if triggered too sensitively._
https://www.nngroup.com/articles/timing-exposing-content/

I'm pushing a change to make delay an optional prop with 0ms as its default value but this should allow it to be set to something else in the cases its needed.

Comment (Locrian24) [reply packages/visual-dev/src/components/Tooltip/Tooltip.tsx]: Having the delay as a prop makes a ton of sense

**Review by Locrian24 — APPROVED** [review]
👍🏻

**Review by noahwc — APPROVED** [review]

**Pull request merged**